tests/meta/test_ensure_type_hints.py

```diff
@@ -4,15 +4,34 @@
 Original notes by John L. Villalovos
 
 """
+import dataclasses
+import functools
 import inspect
-from typing import Tuple, Type
+from typing import Optional, Type
 
 import _pytest
 
 import gitlab.mixins
 import gitlab.v4.objects
 
 
+@functools.total_ordering
+@dataclasses.dataclass(frozen=True)
+class ClassInfo:
+    name: str
+    type: Type
+
+    def __lt__(self, other: object) -> bool:
+        if not isinstance(other, ClassInfo):
+            return NotImplemented
+        return (self.type.__module__, self.name) < (other.type.__module__, other.name)
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, ClassInfo):
+            return NotImplemented
+        return (self.type.__module__, self.name) == (other.type.__module__, other.name)
```

Comment:
TBH I hope we can at some point get rid of all this scaffolding by using more native mypy stuff if it's possible/becomes available 😅